### ISSUE TITLE
eth_getBlockTransactionCountByNumber not counting State Sync

### ISSUE THREAD

**Issue by quickchase** [maticnetwork/bor#564]
It looks like `eth_getBlockTransactionCountByNumber` is not accounting for the State Sync transactions.

Version: `bor/v0.2.16-stable-f083705e`

This also happens on the Erigon `erigon/0.0.3` release with the latest snapshot.

How to reproduce:
```
{"jsonrpc": "2.0", "id": 1, "method": "eth_getBlockTransactionCountByNumber", "params": ["0x20f9b80"]}'
```
Expected Output:
`0x8f` (143)

Actual Output:
`0x8e` (142)

**Comment by 0xKrishna** [comment]
Noted, Will work on it.

**Comment by jiezhang** [comment]
Please don't close it.

**Comment by 0xsharma** [comment]
> @0xsharma Can you please include the fix in the next release?

@jiezhang this will be included in v0.3.1 release.
